### ISSUE TITLE
paket4 install crashes with stack-overflow exception

### ISSUE THREAD

**Issue by cloudRoutine** [fsprojects/Paket#2030]
### Description

I've been using Paket4 in my migration of the netcore port of FAKE, and the parts that work are great, [once you get past some of the initial pitfalls you might stumble into](https://github.com/fsprojects/Paket/issues/2029)

### Repro steps

[paket-install-stackoverflow.zip](https://github.com/fsprojects/Paket/files/601894/paket-install-stackoverflow.zip)

1.  unzip and cd into repo
1. `.\.paket\paket.bootstrapper.exe 4.0.0-alpha024`
1. `.\.paket\paket.exe install`
1. Let it run until 💥 

[ Log of packages installed before stack overflow exception ](https://gist.github.com/cloudRoutine/c15d682028ccfbfbcc7b203ed48b5d57#file-log-ini) 

### Expected behavior

paket installs packages without crashing

### Actual behavior

paket crashes during install

### Known workarounds

haven't found one yet, will update if I do

**Comment by forki** [comment]
weird. can't reproduce.

**Comment by cloudRoutine** [comment]
@forki any ideas about how I could get around this? I ran it 10 times, every time it ended in

![](http://i.imgur.com/3jDmBNt.png)

**Comment by forki** [comment]
can you try to debug it?

**Comment by cloudRoutine** [comment]
@forki  version `4.0.0-alpha030` is still blowing the stack on me

this time on the **paket.dependencies** file 

```
source https://nuget.org/api/v2

github logary/logary src/Logary.Facade/Facade.fs 4b21d888e83d72d1238ab0f36110e2a3946240e7

nuget Argu
nuget FsCheck
nuget FSharp.Core ~> 3
nuget FSharpx.Core
nuget BenchmarkDotNet
nuget NuGet.CommandLine
nuget Microsoft.NET.Sdk alpha 
nuget FSharp.NET.Sdk 1.0.0-alpha-000007 alpha framework: netstandard1.6
nuget NETStandard.Library 1.6.1 framework: netstandard1.6
nuget System.Diagnostics.TraceSource

group netcore 
    source https://www.nuget.org/api/v2

    nuget FSharp.Core 4.0.1.7-alpha alpha framework: netstandard1.6

group build
    source https://www.nuget.org/api/v2

    nuget FAKE
```

![](http://i.imgur.com/ISgQyab.png)

but at least it happens really fast now 😆 
it's gone from 10m down to ~10s

**Comment by forki** [comment]
So I guess I'll see a PR soon? ;-)

Am 30.11.2016 06:16 schrieb "Jared Hester" <notifications@github.com>:

**Comment by cloudRoutine** [comment]
*_Once more unto the breach, dear friends, once more_*

**Comment by cloudRoutine** [comment]
fixed in #2066
